radio/src/mixer.cpp

```diff
@@ -527,13 +527,19 @@ int getStickTrimValue(int stick, int stickValue)
     return 0;
 
   int trim = trims[stick];
-  if (stick == THR_STICK) {
-    if (g_model.thrTrim) {
-      int trimMin = g_model.extendedTrims ? 2*TRIM_EXTENDED_MIN : 2*TRIM_MIN;
-      trim = ((g_model.throttleReversed ? (trim+trimMin) : (trim-trimMin)) * (RESX-stickValue)) >> (RESX_SHIFT+1);
-    }
-    if (g_model.throttleReversed) {
+  uint8_t thrTrimSw;
+  if (g_model.thrTrimSw == 0) //this is needed for compatibility with existing models
+      thrTrimSw = MIXSRC_TrimThr - MIXSRC_FIRST_TRIM;
+    else if (g_model.thrTrimSw == MIXSRC_TrimThr - MIXSRC_FIRST_TRIM)
```

Comment:
I don't get why this else if is needed

Reply:
I've made the assumption that non used eeprom values are 0 (this assumption has been checked with several models on a X7 radio). So  _g_model.thrTrimSw_  will be 0 in existing models when updating opentx from previous releases. And the default switches order is RETA, so T has to go in first place (0) (this is safe as on previous releases throttle trim could only be used correctly on the default throttle switch). This is achieved by interchanging R (2) and T (0). And this is reason of the _else if_

This _else if_ is clearer as:

_else if (g_model.thrTrimSw == 2)_

Reply:
wow, it was impossible to understand!